### ISSUE TITLE
Migration inconsistencies for new install

### ISSUE THREAD

**Issue by archonic** [pay-rails/pay#171]
There's a note about checking CHANGELOG.md for migrations if upgrading from previous versions, so that handles `AddOwnerTypeToPay`, but this migration should be generated by the `rails pay:install:migrations` command for new installs.

Update: It seems this migration was partially rolled into an old migration? I had the owner_type column on pay_charges but not pay_subscriptions, so I had to comment out that line:

```
class AddOwnerTypeToPay < ActiveRecord::Migration[6.0]
  def change
    # add_column :pay_charges, :owner_type, :string
    add_column :pay_subscriptions, :owner_type, :string

    # Backfill owner_type column to match your Billable model
    Pay::Charge.update_all owner_type: "User"
    Pay::Subscription.update_all owner_type: "User"
  end
end
```

Perhaps migrations should be tested? For a fresh install, running through all generated migrations should arrive at a consistent schema.

**Comment by excid3** [comment]
Yeah, it was added to the migration so that new installs won't have to do it. Only existing installs upgrading need to add the migration.

I guess we probably should have had it as an additional migration that gets copied over and just made you run `rails pay:install:migrations` a second time when upgrading.

**Comment by archonic** [comment]
Re-running `rails pay:install:migrations` would be a nice upgrade path.

It looks like that migration was only partially rolled into the old one. I created a new app, wrote a Users migration and followed the install instructions. This was the resulting schema:

```
ActiveRecord::Schema.define(version: 2020_06_19_195127) do

  create_table "pay_charges", force: :cascade do |t|
    t.string "owner_type"
    t.integer "owner_id"
    t.string "processor", null: false
    t.string "processor_id", null: false
    t.integer "amount", null: false
    t.integer "amount_refunded"
    t.string "card_type"
    t.string "card_last4"
    t.string "card_exp_month"
    t.string "card_exp_year"
    t.datetime "created_at"
    t.datetime "updated_at"
  end

  create_table "pay_subscriptions", force: :cascade do |t|
    t.integer "owner_id"
    t.string "name", null: false
    t.string "processor", null: false
    t.string "processor_id", null: false
    t.string "processor_plan", null: false
    t.integer "quantity", default: 1, null: false
    t.datetime "trial_ends_at"
    t.datetime "ends_at"
    t.datetime "created_at"
    t.datetime "updated_at"
    t.string "status"
  end

  create_table "users", force: :cascade do |t|
    t.string "name"
    t.string "email"
    t.datetime "created_at", precision: 6, null: false
    t.datetime "updated_at", precision: 6, null: false
    t.string "processor"
    t.string "processor_id"
    t.datetime "trial_ends_at"
    t.string "card_type"
    t.string "card_last4"
    t.string "card_exp_month"
    t.string "card_exp_year"
    t.text "extra_billing_info"
  end

end
```

It's missing the `owner_type` for pay_subscriptions.

**Comment by cameronbarker** [comment]
Just had the same issue, I updated `t.references :owner` to `t.references :owner, polymorphic: true` in `CreatePaySubscriptions` migration.

**Comment by archonic** [comment]
Looks like it's been fixed in master and just needs releasing.

**Comment by archonic** [comment]
I tried to check if this fix was released and it looks like the rubygems versions and github versions haven't lined up in a while. I'll try to confirm the fix later tonight.

**Comment by excid3** [comment]
Looks like I published 2.2.x from my fork, not the main repo. Whoops. Gonna reconcile those.
